File: src/lib/stores/branding.ts
import { writable } from 'svelte/store';
import { api, type BrandingDto } from '$lib/tauri';

const defaults: BrandingDto = {
  internalId: 'mc-launcher-template', displayName: 'MC Launcher', windowTitle: 'MC Launcher',
  primaryColor: '#7c3aed', secondaryColor: '#1e293b', accentColor: '#f59e0b',
  headingFont: 'Inter', bodyFont: 'Inter', discord: '', website: '',
  serverName: 'Mi Servidor', serverAddress: '',
};

function createBrandingStore() {
  const { subscribe, set, update } = writable<BrandingDto>(defaults);
  return {
    subscribe,
    load: async () => {
      try { set(await api.getBranding()); } catch { /* use defaults */ }
    },
  };
}

export const branding = createBrandingStore();

export function applyBrandingVars(b: BrandingDto) {
  const root = document.documentElement;
  root.style.setProperty('--color-primary',   b.primaryColor);
  root.style.setProperty('--color-secondary', b.secondaryColor);
  root.style.setProperty('--color-accent',    b.accentColor);
  root.style.setProperty('--font-heading',    b.headingFont);
  root.style.setProperty('--font-body',       b.bodyFont);
}

File: src-tauri/src/commands/branding.rs
use serde::Serialize;

const BRANDING_JSON: &str = include_str!(concat!(env!("OUT_DIR"), "/branding.json"));

#[derive(Debug, Serialize, serde::Deserialize, Clone)]
#[serde(rename_all = "camelCase")]
pub struct BrandingDto {
    pub internal_id: String,
    pub display_name: String,
    pub window_title: String,
    pub primary_color: String,
    pub secondary_color: String,
    pub accent_color: String,
    pub heading_font: String,
    pub body_font: String,
    pub discord: String,
    pub website: String,
    pub server_name: String,
    pub server_address: String,
}

#[tauri::command]
pub fn get_branding() -> BrandingDto {
    serde_json::from_str(BRANDING_JSON).expect("branding.json malformed — rebuild")
}
